src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -1487,6 +1485,42 @@ class OptionsSegment(BaseSegment):
     )
 
 
+# ReferenceDefinitionGrammar
+class ForeignKeyReferenceSegment(BaseSegment):
+    """A foreign key reference segment."""
+
+    type = "foreign_key_reference_segment"
+    match_grammar = Sequence(
+        "REFERENCES",
+        Ref("TableReferenceSegment"),
+        Ref("BracketedColumnReferenceListGrammar"),
+        "NOT",
+        "ENFORCED",
+    )
+
+
+class TableConstraintSegment(BaseSegment):
+    """A table constraint segment.
+
+    https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_constraints
+    """
+
+    type = "table_constraint"
```

Comment:
It seems we're inconsistent here and some dialects do what you do and redefine this, while others inherit it from ANSI and override the `match_grammar`.

I've no idea which is better to be honest, or if they are basically equivalent for this simple example? I mean it does save you explicitly setting the `type` (since that's in the base class it inherits from), and I guess it does make it more obvious it's overridden something from ANSI which seems good, but probably not a big deal either way (especially as some dialects do it one way and some another). WDYT?

```suggestion
class TableConstraintSegment(ansi.TableConstraintSegment):
    """A table constraint segment.

    https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_constraints
    """

```

Reply:
I dont mind :)

Reply:
Let's go for the override option. Seems like the better way of doing it.